src/App.tsx

```diff
@@ -17,18 +52,45 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                selectedFilter={selectedFilter}
+                setSelectedFilter={setSelectedFilter}
+                searchQuery={searchQuery}
+                setSearchQuery={setSearchQuery}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {shouldShowError ? (
+              <div className="has-text-danger">
+                Unable to load the todos
+              </div>
+
+            ) : (
+              <div className="block">
+                {isLoading
+                  ? (
+                    <Loader />
+                  ) : (
+                    <TodoList
+                      todos={visibleTodos}
+                      setSelectedTodo={setSelectedTodo}
+                      selectedTodo={selectedTodo}
+                    />
+                  )}
+
+              </div>
+            )}
```

Comment:
looks super difficult. Let's find a solution(s) which would make it easier to read